### PR TITLE
Update Unibeautify and add dependencies property

### LINKED ISSUE #16: Add dependencies property to beautifier
Once https://github.com/Unibeautify/unibeautify/pull/63 is published to npm

- [ ] Update unibeautify dependency
- [ ] Add property `dependencies` to beautifier

```typescript
  dependencies: [
    {
      type: DependencyType.Node,
      name: "ESLint",
      package: "eslint",
    }
  ],
```

### PR DESCRIPTION
**Pull request by szeck87**
Closes #16

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
     "tslint-clean-code": "^0.2.6",
     "tslint-microsoft-contrib": "^5.0.3",
     "typescript": "^2.7.2",
-    "unibeautify": "^0.10.1"
+    "unibeautify": "0.11.1"
```
Comment (Glavin001) [inline package.json]: Should this be a range?

File: src/index.ts [hunk 2]
```diff
@@ -1,11 +1,18 @@
-import { Beautifier, Language, BeautifierBeautifyData } from "unibeautify";
+import { Beautifier, Language, BeautifierBeautifyData, DependencyType } from "unibeautify";
 import * as readPkgUp from "read-pkg-up";
 import options from "./options";
 const CLIEngine = require("eslint").CLIEngine;
```
Comment (Glavin001) [inline src/index.ts]: Remove

File: src/index.ts [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
+  dependencies: [
+    {
+      type: DependencyType.Node,
+      name: "ESLint",
+      package: "eslint",
+    }
+  ],
   options: {
     JavaScript: options.JavaScript,
   },
```
Comment (Glavin001) [inline src/index.ts]: Missing change to https://github.com/Unibeautify/beautifier-eslint/blob/dce2b22afd9362491fff0f4d6702318c9ea64cdb/src/index.ts#L21

Something like:

```typescript
  beautify(data: BeautifierBeautifyData) {
    return new Promise<string>((resolve, reject) => {
       const CLIEngine = data.dependencies.get<NodeDependency>("ESLint").package;
```

Comment (Glavin001) [reply src/index.ts]: See https://github.com/Unibeautify/unibeautify/blob/master/test/DependencyManager/NodeDependency.spec.ts#L58

File: package.json (hunk 1 above)
Comment (szeck87) [reply package.json]: Yes it should, the caret must have been removed during the link.

File: src/index.ts (hunk 3 above)
Comment (Glavin001) [reply src/index.ts]: Whoops, I mean:

```typescript
const { CLIEngine } = data.dependencies.get<NodeDependency>("ESLint").package;
```

File: src/index.ts [hunk 4]
```diff
@@ -1,28 +1,35 @@
...
       });
-      const text = cli.executeOnText(data.text).results[0].output;
-      if (text) {
-        return resolve(text);
+      const result = cli.executeOnText(text).results[0].output;
+      if (result) {
+        return resolve(result);
       } else {
-        return resolve(data.text);
+        return resolve(text);
```
Comment (Glavin001) [inline src/index.ts]: I wonder if we should throw an error here instead? As part of another PR.

**Comment by szeck87** [comment]
@Glavin001 anything more needed here?

**Pull request merged**